woocommerce/woocommerce-gutenberg-products-block
# Add dry run and simulate release mode

This PR adds two modes that allow us to work with safety with deploy scripts.

They have been added two variables. Each variable if it is set to True corresponds to a specific mode.
- `DRY_RUN_MODE`: the commands are just printed, not executed.
- `SIMULATE_RELEASE_MODE`: the commands that affect the local env are executed (e.g. `git commit`), while the commands that affect the remote env are not executed but just printed (e.g. `git push`).


Fixes #3059

### Testing

### Manual Testing

How to test the changes in this Pull Request:

Check out this branch and disconnect the computer from the Internet to avoid unintentional deployment 🚨 

1. Set the variable `DRY_RUN_MODE` to `true` for each script file.
2. Run the command `npm run deploy`.
3. Check that no command is executed.
4. Set the variable `DRY_RUN_MODE` to `false`.
5. Set the variable `SIMULATE_RELEASE` to `true`.
6. Check that only the commands that affect the local env are executed.


Should we add specific commands in the `package.json` to run the deployment script with these different modes?
